File: nzxye/APIServer/models/customer.js
var mongoose = require('mongoose');
var Schema = mongoose.Schema;

var CustomerSchema = new Schema({
  name: {type: String},
  nickName: {type: String},
  sex: {type: String},
  idCard: {type: String},
  mphone: {type: String},
  email: {type: String},
  address: {type: String},
  remark: {type: String},
  created: {type: Date, default: Date.now},
  modified: {type: Date, default: Date.now}
});

// customerSchema.index({name: 1}, {unique: true});
CustomerSchema.index({idCard: 1}, {unique: true});
CustomerSchema.index({email: 1}, {unique: true});
CustomerSchema.index({mphone: 1}, {unique: true});

module.exports = CustomerSchema;

File: nzxye/ScreenAndroid/app/src/main/java/com/nzxye/ai/bean/Customer.java
package com.nzxye.ai.bean;

import com.gustavofao.jsonapi.Annotations.Excluded;
import com.gustavofao.jsonapi.Annotations.Type;
import com.gustavofao.jsonapi.Models.Resource;

/**
 * Created by xiaodongyu on 3/2/17.
 */

@Type("customer")
public class Customer extends Resource {

    private String name;
    private String mphone;
    private String idCard;
    private String sex;
    private String email;
    private String nickName;
    private String address;
    private String remark;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getMphone() {
        return mphone;
    }

    public void setMphone(String mphone) {
        this.mphone = mphone;
    }

    public String getIdCard() {
        return idCard;
    }

    public void setIdCard(String idCard) {
        this.idCard = idCard;
    }

    public String getSex() {
        return sex;
    }

    public void setSex(String sex) {
        this.sex = sex;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getNickName() {
        return nickName;
    }

    public void setNickName(String nickName) {
        this.nickName = nickName;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getRemark() {
        return remark;
    }

    public void setRemark(String remark) {
        this.remark = remark;
    }
}
